src/Surfnet/StepupMiddleware/ApiBundle/Identity/Projector/RaCandidateProjector.php

```diff
@@ -147,14 +253,57 @@ public function applyRegistrationAuthorityRetractedEvent(RegistrationAuthorityRe
             $event->identityInstitution,
             $event->nameId,
             $event->commonName,
-            $event->email
+            $event->email,
+            $event->raInstitution
         );
 
         $this->raCandidateRepository->merge($candidate);
     }
 
-    protected function applyIdentityForgottenEvent(IdentityForgottenEvent $event)
+    /**
+     * @param Institution $institution
+     * @param ConfigurationInstitution[] $authorizedInstitutions
+     * @throws \Doctrine\ORM\NonUniqueResultException
+     */
+    private function updateInstitutionCandidatesFromCollection(Institution $institution, array $authorizedInstitutions)
     {
-        $this->raCandidateRepository->removeByIdentityId($event->identityId);
+
+        $raInstitutions = new InstitutionCollection();
+        foreach ($authorizedInstitutions as $authorizedInstitution) {
+            $raInstitutions->add(new Institution($authorizedInstitution->getInstitution()));
+        }
+
+        $this->raCandidateRepository->removeInstitutionsNotInList($institution, $raInstitutions);
+
+        // loop through authorized institutions
+        foreach ($raInstitutions as $raInstitution) {
+
+            // add new identities
+            $identities = $this->identityRepository->findByInstitution($raInstitution);
+            foreach ($identities as $identity) {
+                $identityId = new IdentityId($identity->id);
+
+                // check if persistent in ra listing
+                if ($this->raListingRepository->findByIdentityIdAndInstitution($identityId, $institution)) {
```

Comment:
The uniqe constraint is on RA institution and identity id, not on the identity institution. This causes problems when event replaying!

Reply:
I fixed this while working on the event replay.